architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+
+#### Wendy tries Radius out
+
+Wendy wants to learn more about Radius, so she fires up a Kubernetes cluster and tries some of our tutorials. She runs `rad init` to install Radius, which includes Dapr and some components that Radius can use for its storage. This is easy for her because she didn't need to learn about Dapr to try Radius out.
+
+*This scenario is here to point out that the simplicity of the current user experience is maintained.*
+
+#### Wendy and Jake put Radius in production
+
+Wendy's been successful experimenting with Radius, and she wants to take it to production. She works with Jake to set up some cloud datastores for Radius to use, to ensure that data is persisted. Jake creates a DynamoDB table and the necessary SNS/SQS resources, following the Dapr documentation. Jake reads the Radius documentation, installs Dapr, and configures the `Component` objects in Kubernetes that are required. Lastly, Jake runs `helm install radius radius -n radius-system` to install Radius without its default component configuration.
+
+
+## User Experience (if applicable)
+
+**Sample Input:**
+
+*n/a*
+
+**rad init Sample Output:**
+
+Running `rad init`:
+
+Before 
+
+```txt
+✅ Installing Radius 0.32.0 on kind-kind
+✅ Create new environment default
+   - Kubernetes namespace: default
+   - Recipe pack: local-dev
+✅ Update local configuration
+```
+
+
+After 
+
+```txt
+✅ Installing Radius 0.32.0 on kind-kind
+✅ Installing Dapr 1.13.0 on kind-kind
+✅ Create new environment default
+   - Kubernetes namespace: default
+   - Recipe pack: local-dev
+✅ Update local configuration
+```
+
+
+**Sample Recipe Contract:**
+
+*n/a*
+
+### Development Experience
+
+*Adding Dapr as a dependency has the potential to impact our development experience. We can follow some existing patterns that Dapr users like to simplify this.*
+
+Wilford is a developer that's making some contributions to Radius for the first time. To get started, he reads the developer document at in [the repo](https://github.com/radius-project/radius/blob/main/docs/contributing/contributing-code/contributing-code-control-plane/running-controlplane-locally.md). He sees that installing Dapr is part of the prerequisites, so he follows the [getting started instructions](https://docs.dapr.io/getting-started/).
+
+After that he follows the rest of the instructions to set up a Kubernetes cluster, and configure his `rad` CLI to talk to the local instance of Radius. When he's done with the setup instructions, he can launch Radius in the debugger. When he does that Radius manages the lifetime of each Dapr process so that Wilford does not to do that manually or learn any extra commands.
+
+## Design
+
+### High Level Design
+
+For our immediate needs we'll be replacing the Radius database and message queue with Dapr's functionality. This doesn't require an architectural change or a large update to the Radius codebase. We'll keep our existing abstractions like the async worker and state store in place. 
+
```

Comment:
would we also eventually use dapr's observability to monitor radius?

Reply:
I think it's an augment not a replacement.